airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLChecker.kt

```diff
@@ -0,0 +1,14 @@
+package io.airbyte.integrations.destination.mssql.v2
+
+import io.airbyte.cdk.load.check.DestinationChecker
+import io.airbyte.integrations.destination.mssql.v2.config.DataSourceFactory
+import io.airbyte.integrations.destination.mssql.v2.config.MSSQLConfiguration
+import jakarta.inject.Singleton
+
+@Singleton
+class MSSQLChecker(private val dataSourceFactory: DataSourceFactory) : DestinationChecker<MSSQLConfiguration> {
```

Comment:
Don't inject the factory -- inject the `DataSource`.  The factory is a Micronaut `@Factory` so it will create the DataSource bean for us.